### PR TITLE
feat: desktop browser provider

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: services/provider/api.go [hunk 1]
```diff
@@ -0,0 +1,432 @@
+package provider
```
Comment (alaibe) [inline services/provider/api.go]: Is provider not too abstract? Maybe web3provider?

File: services/provider/api.go [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+	} else {
+		return api.web3Call(request)
+	}
+
+	/*
+
+		/*
+
+
+		   if data.payload.rpcMethod == "eth_sendTransaction":
```
Comment (alaibe) [inline services/provider/api.go]: i can see we have same technique!

File: services/provider/api_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	}
+}
+
+func setupTestAPI(t *testing.T) (*API, func()) {
+	db, cancel := createDB(t)
+
+	keyStoreDir, err := ioutil.TempDir(os.TempDir(), "accounts")
+
+	// Creating a dummy status node to simulate what it's done in get_status_node.go
+	upstreamConfig := params.UpstreamRPCConfig{
```
Comment (alaibe) [inline services/provider/api_test.go]: so that means you do call for real?

File: services/provider/api.go [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+		MessageID: request.MessageID,
+		Result: JSONRPCResponse{
+			JSONRPC: "2.0",
+			ID:      request.Payload.ID,
+			Result:  result,
+		},
+	}, nil
+}
+
+func (api *API) getVerifiedWalletAccount(address, password string) (*account.SelectedExtKey, error) {
```
Comment (alaibe) [inline services/provider/api.go]: I need to duplicate this function and it is already a copy/paste of an existing function gethbackend. I think we should find a way to not duplicate that. We have an account module already. Shall we move that there?

File: services/provider/api_test.go (hunk 3 above)
Comment (richard-ramos) [reply services/provider/api_test.go]: yeah!

**Review by alaibe — APPROVED** [review]

**Pull request merged**